sse2neon.h

```diff
@@ -6389,6 +6389,11 @@ FORCE_INLINE int _mm_testz_si128(__m128i a, __m128i b)
 #define _mm_extract_epi16(a, imm) \
     vgetq_lane_u16(vreinterpretq_u16_m128i(a), (imm))
 
+// Extract a 16-bit integer from a, selected with imm8, and store the result in
+// the lower element of dst.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_extract_pi16
+#define _mm_extract_pi16(a, imm) vget_lane_u16(vreinterpret_u16_m64(a), (imm))
```

Comment:
Shall we do the type casting?
Since it returns the type `uint16_t` but the returned type of `_mm_extract_pi16(a, imm)` is `int32_t`.

Reply:
Should I use `int` or `int32_t`? It is `int` on official website
```
int _mm_extract_pi16 (__m64 a, int imm8)
```

Reply:
As the description of the document, the result would be 32-bit.
I think we can use `int32_t` without any problem.